corehq/apps/cloudcare/views.py

```diff
@@ -118,7 +118,9 @@ def get_web_apps_available_to_user(self, domain, user):
         ))
         apps = filter(None, apps)
         apps = filter(lambda app: app.get('cloudcare_enabled') or self.preview, apps)
-        apps = filter(lambda app: user.can_access_web_app(domain, app.get('copy_of', app.get('_id'))), apps)
+        # Backwards-compatibility - mobile users haven't historically required this permission
+        if user.is_web_user() or user.can_access_any_web_apps(domain):
```

Comment:
Would it make sense to make this contingent on `WEB_APPS_PERMISSIONS_VIA_GROUPS`?

Reply:
Can you elaborate?  That flag enables the UI for the `app_access.user_can_access_app` filter below, whereas I think this behavior we want for all projects.

Reply:
no, it shouldn't check that flag

Reply:
> Can you elaborate?

I'm suggesting something like

_If the flag is disabled then use the new permissions models, otherwise ignore them and do what it did before._

I think I'm understanding correctly but find it confusing that the flag, when enabled, triggers the _old_ permissions model (via groups) and when disabled triggers the new permissions model via _permissions_?

Reply:
I think your description is right. I'm going to do a PR to access the flag within this function before checking the old style permissions, let's discuss on that PR.

Reply:
https://github.com/dimagi/commcare-hq/pull/34538

Reply:
Re-reading your comment, one adjustment: 
> the new permissions model via permissions instead of groups

The flag does not affect the roles/permissions configuration, so I'd write that as "in addition to", not "instead of"